index.bs

```diff
@@ -1524,6 +1524,132 @@ the interaction with garbage collection.
+
+Choose a dictionary when the part of the API represents data that is transient or you need a configuration-style object or "options bag".
+
+Dictionaries are ideal for when the data doesn't get stored or mutated; it's just used it at the time of the call.
+
+For example, the `ShareData` member from Web Share:
+
+```WebIDL
+dictionary ShareData {
+  USVString title;
+  USVString text;
+  USVString url;
+};
+```
+
+And how it's commonly used:
+
+<pre class="highlight">
+const data = { "text": "Text being shared" };
+await navigator.share(data);
+</pre>
+
+Dictionaries are easily extensible and makes it easy to add optional fields later as needed.
+Members of a dictionary are optional by default, but can be marked as `required` if needed.
+
+Dictionaries are also highly idiomatic (i.e., natural to use in JavaScript). Passing `{ ... }` inline is the most natural way to supply configuration in JavaScript.
+
+Dictionaries, because of how they are treated by user agents, are also relatively future-proof; accommodating new members gracefully, without breaking older code.
+
+Dictionaries are best used for objects that don't need to be distigusied by type in their lifecycle (i.e., `instanceof` checks are mostly meaningless because it's always `Object`).
+
+A key thing to know about dictionries is that they are "passed by value" to methods (i.e., they are copied) and that browsers engines strip unknown members when converting from JavaScript objects to a WebIDL reprensentation.
+This means that if a developer changing the value after it is passed into an API has no effect.
+
+Again, taking the `ShareData` dictinary as an example:
+
+```JS
+const data = {
+    "text": "Text being shared",
+    // Not in the dictionary, so removed by the browser
+    "whatever": 123,
+};
+navigator.share(data);
+
+// Changing this after calling .share() has no effect
+data.text = "New text";
+```
+
+<h4>Choose an Interface for Functionality, State, and Identity</h4>
+
+Intefaces are roughly equivalent to classes in JavaScript. Choose and an interface when a specificaiton need object with data that might also have — or eventually gain — methods, computed or readonly properties, or internal state or "slots".
+
+Unlike dictrionaries, interfaces:
+
+ * provide the ability to check object's identity (i.e., one can check if  it is an `instanceof` a particular class on the global scope),
+ * can have need read-only properties,
+ * can have state,
+ * can exhibit side-effects on assignment.
+
+Defining an interface also exposes it on the global scope, allowing for the specification of static methods. For example, the `canParse()` static method of the URL interface. 
+
+```JS
+if (URL.canParse(someURL)) {
+    // Do stuff...
+}
+```
+
+If the interface can holds data that is useful when serialized (e.g., `GeolocationPosition`), consider adding a `toJSON()` default method. Alternatively, if it holds binary data, one can add .toBlob() an so on.
```

Comment:
Is this toJSON() thing something that belongs here?  Or is it a distraction?

Reply:
Yes, IMO. Because Editors often forget to add this.

Reply:
Maybe a new principle: if it holds data, add a serializer or whatever .toBlob() is classified as.